src/main/kotlin/com/cognifide/gradle/aem/pkg/tasks/PackageConfig.kt

```diff
@@ -0,0 +1,105 @@
+package com.cognifide.gradle.aem.pkg.tasks
+
+import com.cognifide.gradle.aem.AemDefaultTask
+import com.cognifide.gradle.aem.common.file.ZipFile
+import com.cognifide.gradle.aem.common.instance.Instance
+import com.cognifide.gradle.aem.common.instance.service.pkg.Package
+import com.cognifide.gradle.aem.common.instance.service.repository.Node
+import com.cognifide.gradle.aem.common.utils.shortenClass
+import com.cognifide.gradle.common.utils.Patterns
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+open class PackageConfig : AemDefaultTask() {
+
+    /**
+     * Source instance from which OSGi config will be read.
+     */
+    @Internal
+    val instance = aem.obj.typed<Instance> { convention(aem.obj.provider { aem.anyInstance }) }
+
+    /**
+     * Root repository path used to store temporarily OSGi configs for a serialization time only.
+     */
+    @Internal
+    val rootPath = aem.obj.string { convention("/var/gap/package/config") }
+
+    /**
+     * Target directory in which OSGi configs read will be saved.
+     */
+    @Internal
+    val saveDir = aem.obj.dir { convention(aem.packageOptions.configDir) }
+
+    /**
+     * Unique ID of OSGi config to be saved.
+     */
+    @Internal
+    val pid = aem.obj.string { convention(aem.prop.string("package.config.pid")) }
+
+    @TaskAction
+    fun sync() = instance.get().sync {
+        common.progress {
+            step = "Preparing"
+
+            val configPidPattern = pid.orNull ?: aem.javaPackages.joinToString(",") { "$it.*" }
+            val configPids = osgi.determineConfigurationState().pids.map { it.id }.filter { Patterns.wildcard(it, configPidPattern) }
+
+            if (configPids.isEmpty()) {
+                logger.lifecycle("None of OSGi configuration XML files synchronized matching PID '$configPidPattern'!")
+            } else {
+                total = configPids.size.toLong()
+
+                step = "Processing"
+
+                val rootNode = repository.node(rootPath.get())
+                val configNodes = mutableListOf<Node>()
```

Comment:
Avoid using mutable list -> perform mapping

Reply:
considering it, but under the hood, it will be same ;)